lib/eventHubSender.ts

```diff
@@ -350,24 +296,118 @@ export class EventHubSender extends EventEmitter {
   }
 
   /**
-   * Ensures that the token is renewed within the predfiend renewal margin.
+   * Initializes the sender session on the connection.
+   * @returns {Promise<void>}
+   */
+  private async _init(): Promise<void> {
+    try {
+      // Acquire the lock and establish an amqp connection if it does not exist.
+      if (!this._context.connection) {
+        debug(`EH Sender "${this.name}" establishing an AMQP connection.`);
+        await defaultLock.acquire(this._context.connectionLock, () => { return rpc.open(this._context); });
+      }
+
+      if (!this._session && !this._sender) {
+        await this._negotiateClaim();
+        let senderError: any;
+        this._session = await rheaPromise.createSession(this._context.connection);
+        const handleSenderError = (context: rheaPromise.Context) => {
+          senderError = translate(context.sender.error);
+          debug(`An error occurred while creating the sender "${this.name}" : `, senderError);
+        };
+        this._session.on(Constants.senderError, handleSenderError);
+        const options = this._createSenderOptions();
+        debug("Trying to create a sender...");
+        this._sender = await rheaPromise.createSender(this._session, options);
+        debug("Promise to create the sender resolved. Created sender with name: ", this.name);
+        if (senderError) {
+          // There are cases where the EH service sends an attach frame, which causes rhea to emit sender_open event
+          // thus resolving the promise to create a sender and moments later the service sends back a detach frame
+          // indicating that there was some error. Hence we check for senderError, even after the promise has resolved.
+          debug("throwing the senderError, ", senderError);
+          throw senderError;
+        }
+        this._session.removeListener(Constants.senderError, handleSenderError);
```

Comment:
you'll probably still want an error listener internally so you can tear down the link if it dies later on. that way you can set it back up the next time someone tries to send

Reply:
Good point. Will make the required changes.